### PR TITLE
feat: support return_logprob under DSPARK speculative decoding

### PR DESCRIPTION
**Pull request by alphabetc1**
## Motivation

`return_logprob` is rejected under DSPARK, both at admission in the scheduler and again at the top of `DSparkWorkerV2.forward_batch_speculative_generation`.

That rejection is scope, not a technical blocker. #33459 added logprob support for DFLASH: it dropped the `return_logprob` check from `validate_dflash_request` (which covered the whole dflash family) and, because only the DFLASH path had been validated, added a DSPARK-only guard in the scheduler to keep DSPARK where it was. The `raise` inside the DSPARK worker predates that and dates back to the original DSPARK PR (#30261).

The work #33459 did carries over as-is. DFLASH and DSPARK both verify a linear chain, so row `j` of the verify block is the distribution `out_tokens[:, j]` came from, and the tree accept-index gather that `compute_spec_v2_logprobs` expects degenerates to `arange`. DSPARK's `AcceptOuts.out_tokens` has the same `(bs, chain_len)` shape and the same semantics as DFLASH's — rows past the commit length carry unemitted drafts and are sliced off downstream by the accept lengths.

## Modifications

- Extract the chain mapping DFLASH had inline into `compute_spec_v2_chain_logprobs(batch, logits_output, out_tokens)` in `logprob_processor.py`, and call it from both `DFlashWorkerV2` and `DSparkWorkerV2`. DFLASH behavior is unchanged.
- Drop the two DSPARK guards: the `raise ValueError` in the worker, and the scheduler's admission branch (which now falls through to the shared `validate_dflash_request`).
- Mix `SpecLogprobKit` into `TestBasicSanityDSpark` and remove the `@unittest.skip` that stood in for it. Bumped `est_time` 180 → 300.
- Add `test/registered/unit/layers/test_spec_chain_logprobs.py`. The server kits leave one gap: `test_logprob_decode_match_prefill` sends one request at a time, so every request in a batch has the same `top_logprobs_num`, and `test_token_ids_logprob_ragged` only asserts the server does not 500. Nothing checks that heterogeneous per-request logprob params expand in step with the chain row layout — a drift there would silently pair one request's top-k with another request's distribution. The unit test covers that on CPU in ~10s.

## Accuracy Tests

On 1x H200 (`fa3` target + draft backend, Qwen3-14B + `dspark_qwen3_14b_block7`, `block_size=7` / 8 draft tokens, overlap scheduling on, accept length 2.9–3.5):

**DSPARK vs. no-spec baseline.** Same prompt set through both servers, dumping output token ids, per-token logprobs, top-k lists and token-ids logprobs. Cases were 5 x 64-token completions with per-request `top_logprobs_num` of `[5, 0, 3, 5, 1]` and probe id lists of varying length, plus `max_new_tokens` of 1, 2, 3, 5, 6, 7, 8 and 13 so completions end mid-chain and the trailing unemitted draft rows have to be sliced off. The two dumps are byte-identical:

```
$ cmp probe_dspark.json probe_baseline.json
(no output)

DSPARK spec vs no-spec baseline
  worst max|d| = 0.0000; cases with token divergence = 0
```

The decode-vs-prefill self-consistency tables for the two servers also match cell for cell (worst `max|d|` 0.1213, 8 top-1 ties flipped), i.e. the residual is the usual bf16 gap between the decode and prefill paths and is present in the baseline too.

**Test suites.**

- `test_basic_sanity_dspark.py -k logprob` → 6 passed (`test_logprob_decode_match_prefill` reports `max_diff=0.0873 / 0.0792` against a 0.5 threshold).
- `test_basic_sanity_dspark.py` (full) → 28 passed, 1 failed. The failure is `test_basic_math`, which reproduces byte-for-byte on unmodified `main`: Qwen3-14B enters reasoning mode and gets truncated by `max_new_tokens` before emitting `391`. Pre-existing, unrelated to this change.
- `test_dflash.py::TestDFlashServerBase` and `::TestDFlashServerOverlap` `-k logprob` → 12 passed. Regression check for the helper extraction.
- `test_spec_chain_logprobs.py` → 1 passed. Also checked it fails when the row layout is transposed against the param expansion, so it does guard what it claims to.

## Speed Tests and Profiling

No change on the non-logprob path — `compute_spec_v2_chain_logprobs` only runs when `batch.return_logprob` is set, and the DFLASH call site is a straight move of existing code.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

---
### CI States

Latest PR Test (Base): :x: [Run #31879765400](https://github.com/sgl-project/sglang/actions/runs/31879765400)
Latest PR Test (Extra): :x: [Run #31879765275](https://github.com/sgl-project/sglang/actions/runs/31879765275)

### CODE REVIEW

**Comment by alphabetc1** [comment]
already done by https://github.com/sgl-project/sglang/pull/34696
